ThunkLibs/include/common/Host.h

```diff
@@ -111,9 +92,155 @@ struct ParameterAnnotations {
 // Placeholder type to indicate the given data is in guest-layout
 template<typename T>
 struct guest_layout {
+  static_assert(!std::is_class_v<T>, "No guest layout defined for this non-opaque struct type. This may be a bug in the thunk generator.");
+  static_assert(!std::is_union_v<T>, "No guest layout defined for this non-opaque union type. This may be a bug in the thunk generator.");
+  static_assert(!std::is_enum_v<T>, "No guest layout defined for this enum type. This is a bug in the thunk generator.");
+  static_assert(!std::is_void_v<T>, "Attempted to get guest layout of void. Missing annotation for void pointer?");
+
+  using type = std::enable_if_t<!std::is_pointer_v<T>, T>;
+  type data;
+
+  // TODO: Make this conversion explicit
+  guest_layout& operator=(const T from) {
+    data = from;
+    return *this;
+  }
+
+  // Allow conversion of integral types of same size and sign to each other.
+  // This is useful for handling "long"/"long long" on 64-bit, as well as uint8_t/char.
+  // TODO: Make this conversion explicit
```

Comment:
Drop this one. These are constraint anyway, and a future change needs this implicit conversion behavior between "similar" types.

Reply:
Had an epiphany the other day on how to handle this more cleanly. For now, this extra constructor is fine though, but it will be replaced once we land full 32-bit support.

Reply:
Actually this won't be needed at all for a while, so I've dropped it from this PR.